com.unity.shadergraph/Editor/Data/Nodes/Input/Texture/ProceduralVirtualTextureNode.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections.Generic;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Internal;
+
+namespace UnityEditor.ShaderGraph
+{
+#if PROCEDURAL_VT_IN_GRAPH
+    [Title("Input", "Texture", "Procedural Virtual Texture")]
+    class ProceduralVirtualTextureNode : AbstractMaterialNode
+    {
+        public const int OutputSlotId = 0;
+
+        const string kOutputSlotName = "Out";
+
+        public ProceduralVirtualTextureNode()
+        {
+            UpdateNodeAfterDeserialization();
+            SetLayerCount(2);
+
+            vtProperty.displayName = "ProceduralVirtualTexture";
+            vtProperty.overrideReferenceName = "MyPVT";             // TODO : make unique
+            vtProperty.generatePropertyBlock = false;
+            vtProperty.value.procedural = true;
+
+            UpdateName();
+        }
+
+        void UpdateName()
+        {
+            name = "Procedural Virtual Texture: " + vtProperty.overrideReferenceName;
+        }
+
+        public sealed override void UpdateNodeAfterDeserialization()
+        {
+            AddSlot(new VirtualTextureMaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output));
+            RemoveSlotsNameNotMatching(new[] { OutputSlotId });
+            SetLayerCount(layers);
+        }
+
+        [SerializeField]
+        private VirtualTextureShaderProperty vtProperty = new VirtualTextureShaderProperty();
+
+        void SetLayerCount(int layers)
+        {
+            var uniqueName = objectId;
```

Comment:
Hmm what is this objectId?

Reply:
unique number that identifies the node.  Used by a lot of code generation to ensure the names of functions are unique.